diesel/src/connection/mod.rs

```diff
@@ -12,83 +14,77 @@ pub use self::transaction_manager::{AnsiTransactionManager, TransactionManager};
 pub use self::statement_cache::{MaybeCached, StatementCache, StatementCacheKey};
 
 /// Perform simple operations on a backend.
+///
+/// You should likely use [`Connection`](trait.Connection.html) instead.
 pub trait SimpleConnection {
     /// Execute multiple SQL statements within the same string.
     ///
-    /// This function is typically used in migrations where the statements to upgrade or
-    /// downgrade the database are stored in SQL batch files.
+    /// This function is used to execute migrations,
+    /// which may contain more than one SQL statement.
     fn batch_execute(&self, query: &str) -> QueryResult<()>;
 }
 
-/// Perform connections to a backend.
+/// A connection to a database
 pub trait Connection: SimpleConnection + Sized + Send {
-    /// The backend this connection represents.
+    /// The backend this type connects to
     type Backend: Backend;
     #[doc(hidden)]
     type TransactionManager: TransactionManager<Self>;
 
-    /// Establishes a new connection to the database at the given URL. The URL
-    /// should be a valid connection string for a given backend. See the
-    /// documentation for the specific backend for specifics.
+    /// Establishes a new connection to the database
+    ///
+    /// The argument to this method varies by backend.
+    /// See the documentation for that backend's connection class
```

Comment:
Maybe we should override exactly this documentation in the implementation of each Connection

Reply:
It doesn't render in a prominent enough location. I've been toying with the idea of deprecating `Connection::establish` and just having inherent methods for each class (would let us vary the signature too which would be nice)